### ISSUE TITLE
Remove no-longer used keepErrors implementation bits in twisted.python.log

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6510]
|[<img alt="dreid's avatar" src="https://avatars.githubusercontent.com/u/48695?s=50" width="50" height="50">](https://github.com/dreid)| @dreid reported|
|-|-|
|Trac ID|trac#6510|
|Type|enhancement|
|Created|2013-05-08 06:58:45Z|
|Branch|https://github.com/twisted/twisted/tree/kept-errors-6510|

This was added in 2002 [1789] & [1837] to support pre-trial unittest functionality.

Trial stopped using it in 2006 [18352] (and at the same time deprecated the public interface).

The public interface was finally removed in 2012 [34813].

It should be safe to remove the private variables ```_keepErrors```, ```_keptErrors```, and ```_ignoreErrors``` and the corresponding usage in ```log.err```.

<details><summary>Searchable metadata</summary>

```
trac-id__6510 6510
type__enhancement enhancement
reporter__dreid dreid
priority__normal normal
milestone__ 
branch__branches_kept_errors_6510 branches/kept-errors-6510
branch_author__wsanchez wsanchez
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1367996325000000 1367996325000000
changetime__1370264162000000 1370264162000000
version__None None
owner__wsanchez wsanchez
cc__thijs
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

[#6522](https://github.com/twisted/twisted/issues/6522) was a duplicate of this.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez set owner to @wsanchez<br>@wsanchez set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_5"></a>|@wsanchez commented|
|-|-|

(In [38453]) Branch for [#6510](https://github.com/twisted/twisted/issues/6510)

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_6"></a>|@wsanchez set status to `closed`|
|-|-|

(In [38454]) Remove the private variables _keepErrors, _keptErrors, and _ignoreErrors and the corresponding usage in log.err.
Fixes [#6510](https://github.com/twisted/twisted/issues/6510).

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez set status to `reopened`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_8"></a>|@wsanchez commented|
|-|-|

Whoops, Fixes commit closed this prematurely.  My bad.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez set status to `closed`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_10"></a>|@wsanchez set status to `reopened`|
|-|-|

OK so I don’t know Trac so good.  One more time, with feeling.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)|@wsanchez removed owner<br>@wsanchez set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_12"></a>|@therve set owner to @wsanchez|
|-|-|

Awesome. Please add a NEWS fragment and merge.

**Comment by twisted-trac** [comment]
|[<img alt="wsanchez's avatar" src="https://avatars.githubusercontent.com/u/50002?s=50" width="50" height="50">](https://github.com/wsanchez)<a name="note_13"></a>|@wsanchez set status to `closed`|
|-|-|

(In [38577]) Merge branches/kept-errors-6510

Author: wsanchez
Reviewer: therve
Fixes: [#6510](https://github.com/twisted/twisted/issues/6510)

Remove no-longer used keepErrors implementation bits in twisted.python.log.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_14"></a>|@thijstriemstra commented|
|-|-|

This change added a `.misc` and `.imsc` file to topfiles: [twisted/topfiles/6510.imsc](twisted/topfiles/6510.imsc).
